### ISSUE TITLE
Jitter Problem

### ISSUE THREAD

**Issue by BrunoFANG1** [xiaoxiaoxh/reactive_diffusion_policy#6]
I trained my RDP on picking up an egg, but it shakes a lot. Have you run into similar circumstances? 

Here is my inference result recorded from my cell phone.

https://drive.google.com/file/d/1HoijvgzonQpI0QkYdRQudNLyMMOJObSV/view?usp=sharing

Here is my wandb log: 

<img width="2158" alt="Image" src="https://github.com/user-attachments/assets/b04ba73b-9688-4cb4-a753-c2aec3a90740" />

**Comment by xiaoxiaoxh** [comment]
Hi, can you provide the information of the robot type that you are using? In our experiments, we use the trajectory interpolator and the  low-level controller in Flexiv RDK to control the robot with a much higher frequency (>1000hz). For other robots (e.g. So100), you may need to implement a high-frequency trajectory Interpolator and a high-frequency low-level controller to ensure smooth behaviors during inference.

**Comment by BrunoFANG1** [comment]
Arm: ARX L5 https://www.arx-x.com/?product/21.html
Tactile Sensor: 能斯达电子MF0506

Can you post a wandb log for me to compare?

**Comment by ChenWendi2001** [comment]
Here is the training log of RDP (with tactile emb.) in the peeling task. We didn't use a validation set, because the validation loss of a diffusion policy model often fails to reflect actual performance. In our initial experiments, we also observed that although the validation loss was increasing, the model was getting better.

![Image](https://github.com/user-attachments/assets/a6b87ce2-0a7b-4f82-a560-547d3ccda54b)

**Comment by BrunoFANG1** [comment]
Thanks! I reviewed you demo videos and noticed that your results are also full of jitter which makes sense to me, but I don't expect raw diffusion policy should have jitter problem since it output a whole action chunk to execute. Any explanation to it?

**Comment by ChenWendi2001** [comment]
We believe that the jitter in your video is not normal behavior. The jitter in our video is slight, which happens when switching between chunks. I think there might be two reasons for the jitter in your model: one is the inference settings, particularly the **interpolation algorithm** used during inference, and whether a suitable **latency step** has been set. The second is whether there is a bug in the signal processing of the tactile feedback in the fast policy. The interpolation algorithm might be the first thing to check, as manually implementing the sending of a single TCP pose often introduces bugs that result in jitter. We believe that with the correct settings, RDP should exhibit the same level of smoothness as DP.
